src/libexpr/parser.y

```diff
@@ -35,6 +37,8 @@ namespace nix {
         SourcePath basePath;
         PosTable::Origin origin;
         std::optional<ErrorInfo> error;
+        std::vector<Comment> comments;
+        std::map<nix::Expr *, Comment> attachedComments;
```

Comment:
if Comments is a vector, does this mean there can be multiple comments attached to one expr?
We could simplify here because doc-comments are multiline comments only und thus only this is a one-to-one relation instead.